app/models/peer_review.rb

```diff
@@ -13,8 +13,7 @@ class PeerReview < ActiveRecord::Base
   validate :no_students_should_be_reviewer_and_reviewee
 
   def reviewee
-    # TODO - Research optimizing or see if rails can do better
-    Grouping.joins({ submissions: { results: :peer_reviews }}).where('peer_reviews.id = ?', self.id).first
+    result.submission.grouping
```

Comment:
I changed this line as it allowed me to do a nested includes, and it also looks cleaner. Thoughts?
I remember making this on probably my third or fourth day before I knew how to properly use rails, so this might be some ugly stack-overflow hacky thing I did.